Assets/ModScripts/Submodules/SimonSkips.cs

```diff
@@ -0,0 +1,193 @@
+    }
+
+    public override void OnUtilityPress()
+    {
+        if (Module.IsAnimating())
+            return;
+
+        Module.Audio.PlayGameSoundAtTransform(KMSoundOverride.SoundEffect.ButtonPress, Module.transform);
+        Module.UtilityButton.GetComponentInChildren<KMSelectable>().AddInteractionPunch(0.5f);
+
+        if (Module.IsModuleSolved() || Module.Solving)
+            return;
+
+        if (!Module.CheckValidComponents())
+        {
+            Debug.LogFormat("[The Cruel Modkit #{0}] Strike! The ❖ button was pressed when the component selection was [{1}] instead of [{2}].", ModuleID, Module.GetOnComponents(), Module.GetTargetComponents());
+            Module.CauseStrike();
+            return;
+        }
+
+        Module.StartSolve();
+
+        arrowColours = ConvertArrowNumstoLEDNums();
+        NewLEDs();
+        if (!submitEmpty)
+        {
+            FindFullSequence();
+            string[] pressColours = finalSequence.Where(x => x != 8).Select(x => ComponentInfo.ArrowColors[orderedArrows[x]]).ToArray();
+            Debug.LogFormat("[The Cruel Modkit #{0}] The sequence of colours to press is {1} followed by the center button.", ModuleID, string.Join(", ", pressColours));
+        }
+
+        return;
+    }
+
+    public override void OnArrowPress(int Arrow)
+    {
+        if (Module.IsAnimating())
+            return;
+
+        Module.Audio.PlayGameSoundAtTransform(KMSoundOverride.SoundEffect.ButtonPress, Module.transform);
+        Module.Audio.PlaySoundAtTransform(Module.ArrowSounds[Arrow].name, Module.transform);
+        Module.Arrows[Arrow].GetComponentInChildren<KMSelectable>().AddInteractionPunch(0.25f);
+
+        if (Module.IsModuleSolved())
+            return;
+
+        Module.StartCoroutine(HandleArrowFlash(Arrow));
+
+        if (!Module.Solving)
+            return;
+
+        if (Arrow == 8)
+        {
+            inputtedSequence.Add(8);
+            if (inputtedSequence.SequenceEqual(finalSequence))
+            {
+                Debug.LogFormat("[The Cruel Modkit #{0}] The correct sequence has been entered. Solve!", ModuleID);
+                Module.Solve();
+            }
+            else
+            {
+                string[] inputtedColours = inputtedSequence.Where(x => x != 8).Select(x => ComponentInfo.ArrowColors[orderedArrows[x]]).ToArray();
+                Debug.LogFormat("[The Cruel Modkit #{0}] An incorrect sequence of {1} has been submitted. Strike!", ModuleID, string.Join(", ", inputtedColours));
+                Module.CauseStrike();
+                inputtedSequence.Clear();
+            }
+        }
+        else
+        {
+            int[] converter = { 0, 2, 4, 6, 1, 3, 5, 7 };
+            inputtedSequence.Add(converter[Arrow]);
+        }
+    }
+
+    public IEnumerator HandleArrowFlash(int Arrow)
+    {
+        if (Arrow < 0 || Arrow >= 9) yield break;
+        yield return null;
```

Comment:
Is line 187 necessary here? Not sure myself

Reply:
yes it is since you need a yield return inside an IEnumerator otherwise it wont be happy

Reply:
There's already one with the WaitForSeconds though